src/opentracing/instrument/ModuleHack.ts

```diff
@@ -0,0 +1,288 @@
+const Module = require('module');
+const falafel = require('falafel');
+const util = require('util');
+const path = require('path');
+
+const TRACE_ENTRY = 'var __njsEntryData__ = __njsTraceEntry__({file: %s, name: %s, line: %s, args: %s, argNames:%s});';
+// tslint:disable-next-line:max-line-length
+const TRACE_EXIT = '__njsTraceExit__({entryData: __njsEntryData__, exception: %s, line: %s, returnValue: %s, exceptionValue:%s});';
+const ON_CATCH = 'if (__njsOnCatchClause__) {\n__njsOnCatchClause__({entryData: __njsEntryData__});\n}';
+
+import ThundraTracer from '../Tracer';
+import TraceConfig from '../../plugins/config/TraceConfig';
+import TraceOption from '../../plugins/config/TraceOption';
+import {TRACE_DEF_SEPERATOR, ARGS_TAG_NAME, RETURN_VALUE_TAG_NAME } from '../../Constants';
+import Argument from './Argument';
+import ReturnValue from './ReturnValue';
+import Utils from '../../plugins/Utils';
+
+class ModuleHack {
+    tracer: ThundraTracer;
+    traceConfig: TraceConfig;
+
+    constructor(tracer: ThundraTracer, traceConfig: TraceConfig) {
+        this.tracer = tracer;
+        this.traceConfig = traceConfig;
+    }
+
+    shouldTrace(relPath: string): TraceOption {
+        try {
+            if (relPath.includes('node_modules')) {
+                return null;
+            }
+            for (const traceOption of this.traceConfig.traceDef) {
+                const patterns = traceOption.pattern.split(TRACE_DEF_SEPERATOR);
+                const regStr = patterns.slice(0, patterns.length - 1).join().replace(TRACE_DEF_SEPERATOR, '/');
+                const regExp = new RegExp(regStr);
```

Comment:
We should cache `regExp` at somewhere (maybe in the `traceOption`). Pattern should not be parsed every time for trace check.